src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/Mutect2AlleleFilter.java

```diff
@@ -0,0 +1,46 @@
+package org.broadinstitute.hellbender.tools.walkers.mutect.filtering;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.Genotype;
+import htsjdk.variant.variantcontext.VariantContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public abstract class Mutect2AlleleFilter<T> extends Mutect2Filter {
+
+    public Map<Allele, Double> applyFilter(final VariantContext vc, final Mutect2FilteringEngine filteringEngine, ReferenceContext referenceContext) {
+        LinkedHashMap<Allele, List<T>> dataByAllele = vc.getAlleles().stream().collect(Collectors.toMap(Function.identity(), allele -> new ArrayList<>(), (a, b) -> a, () -> new LinkedHashMap<>()));
+        LinkedHashMap<Allele, Double> probabilityByAllele = vc.getAlleles().stream().collect(Collectors.toMap(Function.identity(), allele -> this.getDefaultProbability(), (a, b) -> null, () -> new LinkedHashMap<>()));
+        vc.getGenotypes().stream().filter(filteringEngine::isTumor)
```

Comment:
This is a tricky piece of code and a comment explaining that it's going through all the genotypes to accumulate data for each allele would be  very helpful.